core/src/io/anuke/mindustry/input/InputHandler.java

```diff
@@ -418,17 +419,50 @@ protected void flushRequests(Array<BuildRequest> requests){
                 if(copy.hasConfig && copy.block.posConfig){
                     copy.config = Pos.get(Pos.x(copy.config) + copy.x - copy.originalX, Pos.y(copy.config) + copy.y - copy.originalY);
                 }
+
+                copy.block = didyoumean(req.x, req.y, req.block, req.rotation);
+
                 player.addBuildRequest(copy);
             }
         }
     }
 
+    protected Block didyoumean(int x, int y, Block block, int rotation){
+        Tile over = world.ltile(x, y);
+
+        if(over.block() != null){
+            if(over.block() instanceof Conveyor){
+                if((over.rotation() + rotation) % 2 != 0){
+
+                    Tile behind = over.getNearby((rotation + 2) % 4);
+                    Block previous = behind.block();
+
+                    for(int i = 0; i < lineRequests.size; i++){
```

Comment:
I don't think replacement should work this way. Instead of checking the existing previous block, you could check whether `lineRequests` contains two conveyor requests, one behind and one in front:
```java
lineRequests.contains(r -> r.x == x + Geometry.d4[rotation].x && r.y == y + Geometry.d4[rotation].y && r.block instanceof Conveyor && r.rotation == rotation) && lineRequests.contains(r -> r.x == x - Geometry.d4[rotation].x && r.y == y - Geometry.d4[rotation].y && r.block instanceof Conveyor && r.rotation == rotation)
```